src/components/menus/SnippetList.jsx

```diff
@@ -0,0 +1,32 @@
+import React, { PropTypes } from 'react';
+
+import { MenuItem } from 'material-ui';
+import ArrowDropRight from 'material-ui/svg-icons/navigation-arrow-drop-right';
+
+const makeMenuItems = (userSnippets, onClick) => {
+  if (!Object.keys(userSnippets).length === 0) return null;
+  return Object.keys(userSnippets).map(key => (
+    <MenuItem
+      key={key}
+      primaryText={userSnippets[key].snippetTitle}
+      onClick={() => onClick(key)}
```

Comment:
Essentially you would just pass the onClick prop by itself, and the callback function would take the key (or whatever value it accepts). I'm not going to block on this, though.